### ISSUE TITLE
ImportError even though module is imported

### ISSUE THREAD

**Issue by jakubLangr** [keras-team/keras#6176]
Anyone ever came across this bug: 
```
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-23-cbcfc17289ab> in <module>()
      2 print('Done!')
      3 model_loaded = get_custom_CNN()
----> 4 model_loaded.load_weights('mdl.h5')

/usr/local/lib/python2.7/dist-packages/keras/models.pyc in load_weights(self, filepath, by_name)
    691     def load_weights(self, filepath, by_name=False):
    692         if h5py is None:
--> 693             raise ImportError('`load_weights` requires h5py.')
    694         f = h5py.File(filepath, mode='r')
    695         if 'layer_names' not in f.attrs and 'model_weights' in f:

ImportError: `load_weights` requires h5py.

```
after running this:
```
import h5py # version 2.7.0
print('Done!')
model_loaded = get_custom_CNN()  #this can really be any network that fits mdl.h5
model_loaded.load_weights('mdl.h5')
```
I even checked the exact condition (`h5py is None` is `False`) after seeing the actual code here:
https://github.com/fchollet/keras/blob/master/keras/models.py

Keras `2.0.2`, tf (`1.0.0`) backend. 
Tried:
* Checking for mismatching environments
* Importing h5py from scratch 
* Some old GH tickets suggested installing cython so i did that
* Searching online

**Comment by prernakhurana2** [comment]
Even I am facing the same issue dont know how to resolve.
I have imported h5py despite that its not able to load or save weights.

**Comment by bstriner** [comment]
@jakubLangr Weird. My first guess was the active environment or something but you checked that already. Are you running on current master branch? Maybe there is some weird import error with your h5py.

At the top of models.py it might be eating an error:

```python
try:
    import h5py
except ImportError:
    h5py = None
```

Replace with just `import h5py` or add a `raise` and you might be able to see what is going on.

Alternatively, you might be overwriting h5py somewhere.

Cheers

**Comment by brendanodwyer** [comment]
So the way that i managed to get Keras to run was to add a cell with the below in it.

import h5py as h5py

then just re-run the cells above it after that and it should work.

**Comment by krishnadusad** [comment]
Still have this issue. The 'fixes' here did not work.

**Comment by fedotv** [comment]
Had same issue.
I  reinstalled h5py using pip and restarted the system.
It worked.

**Comment by vmalyi** [comment]
Can't reproduce this issue with Keras (2.0.5) and Python 3.6.1

**Comment by rebeen** [comment]
just install h5py using pipe install h5py

**Comment by Mahedi-61** [comment]
I also has same issue. 
it worked after installing h5py using pip

**Comment by mhs321** [comment]
For me I had to adjust keras.models.py file.

replace line 24-27:

try:
    import h5py
except ImportError:
    h5py = None

with just 

import h5py

make sure you have h5py installed in your environment. Restart your jupyter notebook and should be good to go.

**Comment by U-C-J** [comment]
When I first got this error, I use pip installed h5py, and then I still got this error. After reading all the comments above, I just closed my spyder IDLE and anaconda and reopened, then it works.

**Comment by gontthias** [comment]
good solution

**Comment by clockzhong** [comment]
I've installed h5py with "pip install h5py", but the same issue still happen.

**Comment by G-Hung** [comment]
**TLDR:** 
if you tried pip install h5py many times already
try: pip install keras==2.1.2


I faced the same issue when I tried to use keras in google datalab and tried almost every method that I can google (mainly reinstall h5py in different ways) but it doesn't work

At last when I almost gave up, I found an old notebook back to December where my keras version was 2.1.2 and I just reinstall that version and it goes well for now

**Comment by jakubLangr** [comment]
That seems good enough.

**Comment by rmahfuz** [comment]
As suggested, I uninstalled both keras and h5py, and then installed keras first and then h5py. That did not resolve the issue. I also tried commenting out the try-except as suggested by @mhs321, and only left the import statement. In spite of that, I still get the error 'cannot import name load_weights'. Strangely, it can import load_model. Any help will be appreciated!

**Comment by Elektriman** [comment]
for me, the command `conda install h5py` did not actually install the package on my active conda env **even thought the output said the installation was successfull**.

A way to check if h5py is properly installed is using `conda list` to check if the package is there.

For this special case, you can go to `C:\Users\[username]\Anaconda3\envs\[environment_name]\Lib\site-packages\tensorflow\python\keras\saving\save.py` and find this at the top :
```py
try:
  import h5py
except ImportError:
  h5py = None
  ```
you can try to raise the *ImportError* and it should give you `ImportError: No module named h5py`

**SOLUTION** :
I had to go to my Anaconda navigator and manually add h5py to my environment.
